google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/MavenBuildDeploymentSource.java

```diff
@@ -127,17 +115,17 @@ public File getFile() {
         new File(mavenProject.getBuildDirectory()).getPath() + File.separator
             + mavenProject.getFinalName();
 
-    XmlFile appEngineWebXml = AppEngineAssetProvider.getInstance()
-        .loadAppEngineStandardWebXml(project, Collections.singletonList(getModule()));
     AppEngineProjectService projectService = AppEngineProjectService.getInstance();
 
     // The environment will be null for newly deserialized deployment sources to ensure freshness.
     // In this case, we need to reload the environment.
     if (environment == null) {
-      environment = projectService.getModuleAppEngineEnvironment(appEngineWebXml);
+      environment = projectService.getModuleAppEngineEnvironment(getModule()).orElseThrow(
+          () -> new RuntimeException("No environment.")
```

Comment:
Just thinking this through:

The only way now for this to return Optional.Empty and throw the exception is if there is a MavenBuildDeploymentSource, but there is no facet (due to your changes to getModuleAppEngineEnviornment).

But the user can still create a deployment without the flex framework / facet. However, this code path should _still_ not happen because under these circumstances the maven deployment source wouldn't have been available in the first place. Does this sound correct?